pynumaflow/info/info_types.py

```diff
@@ -0,0 +1,58 @@
+from enum import Enum
+from typing import Dict
+
+# Constants for using in the info-server
+# Need to keep consistent with all SDKs and client
+SERVER_INFO_FILE_PATH = "/var/run/numaflow/server-info"
+EOF = "U+005C__END__"
+
+# Env variables to be passed in the info server metadata.
+# These need to be accessed in the client using the same key.
+# Format - (key, env_var)
+metadata_envs = [("CPU_LIMIT", "NUMAFLOW_CPU_LIMIT")]
+
+
+class Protocol(str, Enum):
+    """
+    Enumerate grpc server connection protocol.
+    """
+
+    UDS = "uds"
+    TCP = "tcp"
+
+
+class Language(str, Enum):
+    """
+    Enumerate Numaflow SDK language.
+    """
+
+    GO = "go"
+    PYTHON = "python"
+    JAVA = "java"
+
+
+class ServerInfo:
+    """
+    ServerInfo is used for the gRPC server to provide the information such as protocol,
+    sdk version, language, metadata to the client.
+
+    Args:
+        protocol: Protocol to use (UDS or TCP)
+        language: Language used by the server(Python, Golang, Java)
+        version: Numaflow sdk version used by the server
+        metadata: Any additional information to be provided (env vars)
+    """
+
+    def __init__(
+        self,
+        protocol: Protocol = None,
+        language: Language = None,
+        version: str = None,
+        metadata: Dict = None,
```

Comment:
After being converted to JSON and persisted to file, is python `Dict` the same format as Java Map and Golang map?

Have you compared the final Json string with java/golang serverInfo? I think we need to make sure same ServerInfo object is written in same Json regardless of which SDK, otherwise we might potentially get parsing error.

Reply:
In my testing I'm able to see the golang client parsing the python info-server file properly
```
% kubectl logs simple-pipeline-mult-0-2q7uj numa | grep Serv
2023/04/21 22:25:49 ServerInfo: &{tcp python 0.4.0 map[CPU_LIMIT:6]}
```